python/lsst/daf/butler/registries/oracleRegistry.py

```diff
@@ -67,10 +70,20 @@ def setConfigRoot(cls, root, config, full):
     def __init__(self, registryConfig, schemaConfig, dimensionConfig, create=False,
                  butlerRoot=None):
         registryConfig = SqlRegistryConfig(registryConfig)
+
+        # currently code in class specific to cx_oracle driver
+        url = engine.url.make_url(registryConfig[".db.url"])
+        if url.get_driver_name() != "cx_oracle":
+            raise ValueError("Unsupported driver(%s).  cx_oracle is only driver supported by OracleRegistry" %
+                             url.get_driver_name())
+
         self.schemaConfig = schemaConfig
         super().__init__(registryConfig, schemaConfig, dimensionConfig, create,
                          butlerRoot=butlerRoot)
 
+        if ".db.schema" in self.config:
```

Comment:
I see that the documentation for `OracleRegistry` is wrong in that it doesn't document all the parameters that are accepted by the `__init__` method. Regardless, can you please document somewhere what db fields Oracle uses?

Reply:
How do you specify Oracle URL? Can this `db.schema` be extracted from URL itself?

Reply:
There does not currently appear to be any SQLAlchemy/cx_Oracle way to specify schema on the URL.   We could put the schema on the URL string and pre-parse it out ourselves before giving it to the create_engine.   As discussed in the Gen3 meeting, we are trying to move towards separating common server information from user information to have a central config + smaller user config.   The dict approach will allow us to use existing config combining methods.

Reply:
OK, sounds reasonable. I was worried a bit that we add backend-specific information to a generic configuration but schemas are supported in Postgres too so this option could be more generic that just Oracle. One thing I'm still not very happy about is a sort of backdoor-ish way of communicating schema name to the `Metadata` (via `self._schema.metadata.schema`). It is probably cleaner to pass `schema=` argument to `Metadata` constructor if we could.

Reply:
The call to the Metadata constructor is buried farther down in code.  oracleRegistry.__init__ -> sqlRegistry.__init__ -> sqlRegistry._createSchema -> Schema.__init__ (doesn't get passed db config) -> SchemaBuilder.__init__ -> Metadata.

I could change the Schema codes to also get the db config.   The code could just assume that if db.schema is defined it should set it and let SQLAlchemy die if DB type doesn't like schema (so there wouldn't have to be dialect checks or Oracle specific schema.py).  This just seemed to be more invasive to the core code for the current ticket.

Reply:
Andy and I had a longer Slack discussion in which he said:  

>  I think it should be better this way, if Metadata is instantiated with incorrect (empty) schema name first there is a chance that that schema name is used for some operations before you change it to a correct one and that can cause issues.